src/components/builder/comparison-value-input.tsx

```diff
@@ -0,0 +1,60 @@
+import {useFormikContext} from 'formik';
+import {ExtendedComponentSchema, Utils as FormioUtils} from 'formiojs';
+import {useContext, useEffect, useState} from 'react';
+import usePrevious from 'react-use/esm/usePrevious';
+
+import {BuilderContext} from '@/context';
+import {getRegistryEntry} from '@/registry';
+import Fallback from '@/registry/fallback';
+import {comparisonValueComponentProps} from '@/registry/types';
+
+export interface simpleConditional {
+  conditional: {
+    eq: string;
+    when: string;
+    show: null | boolean;
+  };
+}
+
+export const ComparisonValueInput: React.FC = () => {
+  const {getFormComponents} = useContext(BuilderContext);
+  const {values, setFieldValue} = useFormikContext<simpleConditional>();
+
+  const [chosenComponent, setChosenComponent] = useState<ExtendedComponentSchema | null>(null);
+  const previousWhen = usePrevious(values.conditional.when);
+
+  useEffect(() => {
+    const conditional = values.conditional;
+    if (conditional?.when === '' && conditional?.eq !== '') {
+      setFieldValue('conditional.eq', '');
+      setChosenComponent(null);
+      return;
+    }
+
+    if (previousWhen !== conditional?.when) {
+      setFieldValue('conditional.eq', '');
+    }
+
+    const component: ExtendedComponentSchema = FormioUtils.getComponent(
+      getFormComponents(),
+      conditional.when,
+      false
+    );
+
+    setChosenComponent(component);
+  }, [values, setFieldValue]);
+
+  if (!chosenComponent) return null;
+
+  const registryEntry = getRegistryEntry(chosenComponent);
+  const {comparisonValueComponent} = registryEntry;
+  const ComparisonValueInputComponent =
+    comparisonValueComponent || Fallback.comparisonValueComponent;
```

Comment:
I think we don't need this layer of going through the registry at all, instead, why don't we pass the `Component` as a prop in the `EditForm`?

E.g. for number/currency:

```tsx
const EditForm = () => {
  // hooks and stuff
  return (
    <Tabs>
      <TabList />

      {/* Advanced tab */}
      <TabPanel>
        <SimpleConditional valueAs={NumberField} />
      </TabPanel>

      {/* ... rest of tab content */}
    </Tabs>
  );
};
```

so then you can easily keep the default value of `valueAs` prop to the `TextField`, which keeps it backwards compatible, and you update the currency/number edit forms.

Advantage is that we don't have to expand the interface of the `RegistryEntry` (particularly with optional properties - with TS I prefer to make as much as possible mandatory because it simplifies reasoning about things), and we keep the same drawback: you have to remember to specify the correct field type for the value to be rendered.

Alternatively, instead of passing a component "type", you could also accept a render function (which is more akin to best practices):

```tsx
<SimpleConditional renderValue={valueProps => <NumberField {...valueProps} />} />
```
which in itself is nice because you can manage the `multiple` at this point instead of generically, so I think this would have my preference even :thinking:

Reply:
I don't think we can pass the component in the edit form as prop? Maybe I am misunderstanding what you mean.

In the edit form, we don't know in advance which component will be selected in the condition. => We use the type of the component which is selected in the conditional tab, not the type of the component corresponding to the EditForm we are in.

Reply:
Ah yes, I was confusing the type of the component we're setting up with the constraint for the eq value, but that doesn't hold up, since you need the data type of the reference component. And the same applies for the `multiple` aspect.

Multiple I find tricky in general because File field for example has different `multiple` behaviour, but that's out of scope for here now.

Keep it as is, I was wrong!